### ISSUE TITLE
batchsize 

### ISSUE THREAD

**Issue by juzhongren** [seobeomjin/ML_Project_MRI_Brain_Image_SR#1]
Excuse me, I would like to ask, what graphics card are you using on google colab, what is the size of the video memory, I use the free one on colab, the video memory is 16G, and the batchsize can only be set to 1

**Comment by juzhongren** [comment]
your loss = L1_loss + self.lmda * D_loss is wrong,should  loss = L1_loss + self.lmda * G_loss

**Comment by seobeomjin** [comment]
Hi, thank you to check this repository

1. I also used a free one from google colab.
2. Thank you for your check. The variable ‘loss’ to collect statistical information was wrong I think. But the training process would not have been a problem because networks were trained by each G_loss and D_loss term.

Thanks.

**Comment by juzhongren** [comment]
Another problem is the problem of gradient calculation,
def updated(self, lr_patches, hr_patches):
         #forward
         for p in self.netG.parameters():
             p.requires_grad = True
It feels like it should be changed to false, and the gradient of the generator should not be calculated.
         gradient_penalty = self.calc_gradient_penalty(hr_patches,sr_patches)
         gradient_penalty.backward(retain_graph=True)
         D_loss += gradient_penalty
         # Semi-supervised Loss (main loss)
         #L1_loss from generator and D_loss from Discriminator as wgan-gp
         loss = L1_loss + self.lmda * D_loss
         # backward + optimize only if in training phase
         D_loss.backward(retain_graph=True)
         self.optimizerD.step()
When calculating the discriminator here, the gradient will accumulate when the gradient is penalized, and the calculation graph can be released to save video memory.
